src/dbinterface.jl

```diff
@@ -384,11 +384,22 @@ Base.IteratorEltype(::Type{Cursor{false, knownlength}}) where {knownlength} = Ba
 Base.eltype(x::Cursor{false, knownlength}) where {knownlength} = Row
 
 function Base.iterate(x::Cursor{false}, st=1)
-    status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
-    while status == API.SQL_STILL_EXECUTING
+    status = API.SQL_SUCCESS
+    while true
         status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
+
+        status == API.SQL_STILL_EXECUTING   && continue
+        status == API.SQL_SUCCESS           && break
+        status == API.SQL_NO_DATA           && return nothing
+        status == API.SQL_NTS               && return nothing
```

Comment:
This is based on the previous behaviour, which may be hiding a small bug. Not sure what we need to do if we get this (and I haven't stumbled upon it)